modules/nf-commons/src/main/nextflow/plugin/HttpPluginRepository.groovy

```diff
@@ -0,0 +1,140 @@
+package nextflow.plugin
+
+import com.google.gson.Gson
+import dev.failsafe.Failsafe
+import dev.failsafe.FailsafeExecutor
+import dev.failsafe.Fallback
+import dev.failsafe.RetryPolicy
+import dev.failsafe.event.EventListener
+import dev.failsafe.event.ExecutionAttemptedEvent
+import dev.failsafe.function.CheckedSupplier
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import nextflow.BuildInfo
+import org.pf4j.update.FileDownloader
+import org.pf4j.update.FileVerifier
+import org.pf4j.update.PluginInfo
+import org.pf4j.update.SimpleFileDownloader
+import org.pf4j.update.verifier.CompoundVerifier
+
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+/**
+ * Represents an update repository served via an HTTP api.
+ */
+@Slf4j
+@CompileStatic
+class HttpPluginRepository implements PrefetchUpdateRepository {
```

Comment:
it would be nice to be better describe how this prefect logic works. Where the pre-fetch data is stored. How it's loaded, when it expires, etc.